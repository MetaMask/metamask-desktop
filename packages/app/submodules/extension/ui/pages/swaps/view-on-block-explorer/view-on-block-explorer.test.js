import React from 'react';

import { renderWithProvider, fireEvent } from '../../../../test/jest';
import ViewOnBlockExplorer from '.';

const createProps = (customProps = {}) => {
  return {
    txHash:
      '0x58e5a0fc7fbc849eddc100d44e86276168a8c7baaa5604e44ba6f5eb8ba1b7eb',
    blockExplorerUrl: 'https://etherscan.io',
    ...customProps,
  };
};

describe('ViewOnBlockExplorer', () => {
  it('renders the component with initial props', () => {
    const { container, getByText } = renderWithProvider(
      <ViewOnBlockExplorer {...createProps()} />,
    );
    expect(getByText('View Swap at etherscan.io')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('clicks on the block explorer link', () => {
    global.platform = { openTab: jest.fn() };
    const { getByText } = renderWithProvider(
      <ViewOnBlockExplorer {...createProps()} />,
    );
    const link = getByText('View Swap at etherscan.io');
    expect(link).toBeInTheDocument();
    fireEvent.click(link);
    expect(global.platform.openTab).toHaveBeenCalledWith({
      url: 'https://etherscan.io',
    });
  });
});
