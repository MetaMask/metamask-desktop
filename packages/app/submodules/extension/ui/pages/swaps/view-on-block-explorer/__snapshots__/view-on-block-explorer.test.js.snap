// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewOnBlockExplorer renders the component with initial props 1`] = `
<div>
  <div
    class="box view-on-block-explorer box--margin-top-6 box--flex-direction-row"
  >
    <button>
      View Swap at etherscan.io
    </button>
  </div>
</div>
`;
